src/main/java/net/io/kino/service/impl/ReservationServiceImpl.java

```diff
@@ -25,10 +34,40 @@
 
     @Override
     public Order createOrder(List<Ticket> tickets, PersonalDetails client) {
-        Order order = new Order(tickets, client, OrderState.inProgress);
+        Order order = new Order(tickets, client, OrderState.inProgress, LocalDateTime.now());
         return orders.save(order);
     }
 
+    @Override
+    public boolean verifyOrder(Order order) {
+        OrdersGetRequest request = new OrdersGetRequest(String.valueOf(order.getTransactionId()));
+        HttpResponse<com.paypal.orders.Order> response = null;
+        try {
+            response = client().execute(request);
+        } catch (IOException e) {
+            e.printStackTrace();
+            return false;
+        }
+
+        double paid = 0;
+        for (PurchaseUnit pu : response.result().purchaseUnits()) {
+            paid += Double.parseDouble(pu.amount().value());
+        }
+        double orderValue = 0;
+        for (Ticket t : order.getTickets()) {
+            orderValue += t.getTicketType().getPrice();
+        }
+
+        if (response.result().status().equals("COMPLETED") && paid == orderValue) {
```

Comment:
Nie jestem pewny czy takie porównywanie Doubli zadziała w każdym przypadku ze względu na precyzję. Może jakiś BigDecimal?

Reply:
Wiecie co, zróbcie transakcje po stronie serwera i nie będzie tego checka. 
https://developer.paypal.com/docs/checkout/reference/server-integration/set-up-transaction/

Trochę dziwi mnie, że w dokumentacja paypala nie mówi zbytnio jak te transakcje weryfikować i sugeruje w początowych działach tworzenie płatności w jsie, a nigdzie nie mówi jak to zweryfikować

Reply:
Chyba w createOrder możecie to wykonać tylko podstawcie do Ordera to id co dostaniecie z paypala